lib/zclient.h

```diff
@@ -405,6 +414,10 @@ struct zapi_nexthop {
 	uint8_t label_num;
 	mpls_label_t labels[MPLS_MAX_LABELS];
 
+	/* SRv6 localsid info for Endpoint-behaviour */
+	uint32_t seg6local_action;
+	struct seg6local_context seg6local_ctx;
+
```

Comment:
The sorting of zapi_nexthops must match the sorting of nexthops - I think the nexthop struct's version of this is included in the nexthop sort, so these should be added to the zapi nexthop sort code.

Reply:
Thank you for your comment 👍 

Ok, so I should fix like a following.
```
AS-IS
struct zapi_nexthop {
  int a;
  uint32_t seg6local_action;
  struct seg6local_content seg6local_ctx;
  int b;
}
struct nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content *seg6local_ctx;  
}
```

```
TO-BE
struct zapi_nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content seg6local_ctx;
}
struct nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content *seg6local_ctx;  
}
```

I'll just fix it.

Reply:
Thank you for your comment!

You are talking about following lines, right?
https://github.com/FRRouting/frr/blob/92b4f625d53df3b29a8b5075029b031cc4016e8b/lib/nexthop_group.c#L349-L354

If yes, I need to update around following lines, don't you.
https://github.com/FRRouting/frr/blob/92b4f625d53df3b29a8b5075029b031cc4016e8b/lib/nexthop.c#L192-L203

I'll just fix it.

Reply:
I'll push following update.
```diff
diff --git a/lib/nexthop.c b/lib/nexthop.c
index 36a02c6..91828cc 100644
--- a/lib/nexthop.c
+++ b/lib/nexthop.c
@@ -66,6 +66,37 @@ static int _nexthop_labels_cmp(const struct nexthop *nh1,
        return memcmp(nhl1->label, nhl2->label, nhl1->num_labels);
 }

+static int _nexthop_seg6local_cmp(const struct nexthop *nh1,
+                                 const struct nexthop *nh2)
+{
+       enum seg6local_action_t act1;
+       enum seg6local_action_t act2;
+       struct seg6local_context *ctx1;
+       struct seg6local_context *ctx2;
+
+       act1 = nh1->nh_seg6local_action;
+       act2 = nh2->nh_seg6local_action;
+       ctx1 = nh1->nh_seg6local_ctx;
+       ctx2 = nh2->nh_seg6local_ctx;
+
+       if (act1 > act2)
+               return 1;
+
+       if (act1 < act2)
+               return -1;
+
+       if (!ctx1 && !ctx2)
+               return 0;
+
+       if (ctx1 && !ctx2)
+               return 1;
+
+       if (!ctx1 && ctx2)
+               return -1;
+
+       return memcmp(ctx1, ctx2, sizeof(struct seg6local_context));
+}
+
 int nexthop_g_addr_cmp(enum nexthop_types_t type, const union g_addr *addr1,
                       const union g_addr *addr2)
 {
@@ -199,6 +230,10 @@ int nexthop_cmp(const struct nexthop *next1, const struct nexthop *next2)
                return ret;

        ret = _nexthop_labels_cmp(next1, next2);
+       if (ret != 0)
+               return ret;
+
+       ret = _nexthop_seg6local_cmp(next1, next2);

        return ret;
 }```

Reply:
And I should fix like a following.
```AS-IS
struct zapi_nexthop {
  int a;
  uint32_t seg6local_action;
  struct seg6local_content seg6local_ctx;
  int b;
}
struct nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content *seg6local_ctx;  
}```
```TO-BE
struct zapi_nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content seg6local_ctx;
}
struct nexthop {
  int a;
  int b;
  uint32_t seg6local_action;
  struct seg6local_content *seg6local_ctx;  
}```
I’ll just fix it.

Reply:
This is my force-push according to your great review. Thank you so much.
https://github.com/FRRouting/frr/compare/14b17e4d9b831a23e78612614bb20d23d8db825e..92eca9a4f0f8532f301a8a824238adc3d11ec4d1

Reply:
And the reason why I can’t found this miss take, is that test case isn’t enough. Sharpd should support the seg6 nexthop like a label/lsp. But supporting it on sharpd is little-bit more work, It’s future work todo. Then I created the issue.

https://github.com/FRRouting/frr/issues/6964